### ISSUE TITLE
How to detect user interaction in any part of ImGui?

### ISSUE THREAD

**Issue by crazydef** [ocornut/imgui#7012]
Version: 1.89.4
Branch: master
Back-ends: Custom renderer
Compiler: VS2022
Operating System: Windows

**My Issue/Question:**

How can I tell if any ImGui window/component/control within the current context is being interacted with by the user?

I'm currently using this:

```
if( !ImGui::IsWindowHovered(ImGuiHoveredFlags_AnyWindow) && !ImGui::IsAnyItemActive() && !ImGui::IsAnyItemFocused())
{
   ...
}
```

but this test fails when the mouse is over any part of a window that has a combo box with an open menu.

**Screenshots/Video**
[ImGuiBug.zip](https://github.com/ocornut/imgui/files/13363219/ImGuiBug.zip)

**Comment by ocornut** [comment]
> How can I tell if any ImGui window/component/control within the current context is being interacted with by the user?

In order to get a good answer you will need to explain exactly why you need this and what are you going to do with information.

**Comment by crazydef** [comment]
I need to know whether mouse movement/clicks are being handled by ImGui or whether I can use them for non-UI interactions.

**Comment by ocornut** [comment]
You should simply read `io.WantCaptureMouse`:
https://github.com/ocornut/imgui/blob/master/docs/FAQ.md#q-how-can-i-tell-whether-to-dispatch-mousekeyboard-to-dear-imgui-or-my-application

- It is in the FAQ.
- The input handler of all our 24 examples mentions it.
- It's in imgui.cpp "how a simple application may look like" section
- It's answered in imgui.cpp simplified copy of the FAQ (which has links for most answered but this one is spelled out).
- It's mentioned in demo's "Inputs & Focus" section.
- In imgui.h the description for IsWindowHovered() says "NB: If you are trying to check whether your mouse should be dispatched to imgui or to your app, you should use the 'io.WantCaptureMouse' boolean for that! Please read the FAQ!"
- In imgui.h near the hovered flags it says "Note: if you are trying to check whether your mouse should be dispatched to Dear ImGui or to your app, you should use 'io.WantCaptureMouse' instead! Please read the FAQ!".

I'm genuinely sincerely asking, what would have made you find this by yourself?
What better thing can I do to reduce the amount of people asking a wrong question leading to this answer?
Any help welcome.

**Comment by crazydef** [comment]
Thanks.

To be honest, I don't generally read documentation outside of source code. (Partly because I don't know where it is, and partly because I just generally assume (for historical reasons) it's incomplete and/or out of date.) We have dozens of third party libraries in our code base, so trying to find online documentation is often more trouble than it's worth.

Hitting F12 on an ImGui function in Visual Studio will take me into ImGui.cpp (or something similar), and that's where I would prefer things to be documented. If every function had a detailed description of what it does, when/why I need to call it, and/or, more importantly, why I shouldn't be calling it if there's something better, I wouldn't have asked a question here.

The comments in ImGui.h are too random and brief.  When I do end up in the header file, there's maybe a 50/50 chance that the documentation there is actually useful, so I more often than not don't bother.  (Although I see in this particular instance, I would have found partially what I was looking for.)  But saying "please read the FAQ" doesn't help much.  Include a link to the exact section that is relevant for that comment.)

I would say a FAQ isn't really real documentation though.  Having a comprehensive online API reference manual for each function  that is indexed by Google would probably be the most useful reference for the majority of people.  As it stands, most Google searches bring me to these issue pages where others have had the same question already answered.  :)  (But, obviously, that wouldn't be necessary if each function was documented in the source code.)

**Comment by ocornut** [comment]
I have pushed several tweaks following your feedback.

> To be honest, I don't generally read documentation outside of source code.

6 out of 7 references I posted are in the source code.

> Hitting F12 on an ImGui function in Visual Studio will take me into ImGui.cpp (or something similar),

It takes me to imgui.h.

>  If every function had a detailed description of what it does, when/why I need to call it, and/or, more importantly, why I shouldn't be calling it if there's something better, I wouldn't have asked a question here.

In f298491 I now have copied the comments from the .h at the top of the .cpp section of IsWindowHovered()/IsAnyItemHovered() since they are the most common incorrect targets for people trying to solve this. I reckon this should have a good effect on this.

> The comments in ImGui.h are too random and brief. When I do end up in the header file, there's maybe a 50/50 chance that the documentation there is actually useful,

I don't agree they are too random, majority of comments were added exactly based on feedback like yours. a 50/50 chance that the doc is useful to your case is pretty good to me.
The underlying real problem is that when you add too many comments/docs then people are even less likely to read them and find what they need. Everyone thinks their question should be a FAQ. But that specific question, it _IS_ a FAQ, it's been asked hundreds of times. If I input exactly your subject line into Google the first match gives the correct answer. If I input exactly your first line below the subject into Google, I also get the answer in the first match. The plain truth is that people aren't willing to read much, they are expecting free tailored service.

Writing cohesive and complete documentation is really difficult. Every function has dozens of subtle or undefined behaviors depending on the exact situations, flags combination, etc. and attempting to document them can be backfiring as you start getting people to rely on them and suddenly implementation details for things that haven't been thoroughly designed yet are becoming API standards. I am afraid I don't have a perfect solution to this problem. But you would admit it is frustrating that people aren't reading. I'm doing my best at creating a web of cross-references, pretty much all comments, wiki, faq pages are up to date (apart from some a few wiki page attempts by people who lose interest after an hour, which I am sometimes marking as obsolete in bold or deleting).
